src/cryptography/hazmat/primitives/serialization/pkcs7.py

```diff
@@ -328,6 +409,15 @@ def _smime_enveloped_encode(data: bytes) -> bytes:
     return m.as_bytes(policy=m.policy.clone(linesep="\n", max_line_length=0))
 
 
+def _smime_enveloped_decode(data: bytes) -> bytes:
+    m = email.message_from_bytes(data)
+    # Content type can be application/x-pkcs7-mime or application/pkcs7-mime
+    # Could also be using the filename parameter, or no checks at all.
+    if "pkcs7-mime" not in m.get_content_type():
```

Comment:
Let's explicitly check for those two, not allow anything that happens to have this substring.

Reply:
Done!